cognite/neat/_session/_session.py

```diff
@@ -0,0 +1,35 @@
+from cognite.neat._utils.http_client._client import HTTPClient
+
+from ._state_machine import EmptyState, ForbiddenState, State
+
+
+class NeatSession:
+    """A session is an interface for neat operations. It works as
+    a manager for handling user interactions and orchestrating
+    the state machine for data model and instance operations.
+    """
+
+    def __init__(self, client: HTTPClient | None = None):
```

Comment:
Lets keep the client there for now, but I think we will replace it by a config before we release this to the end-user.